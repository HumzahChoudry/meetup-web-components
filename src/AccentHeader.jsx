import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

export const ALIGN_VALUES = ['left', 'right', 'center'];

const AccentHeader = ({ align, headingTag, children, className, ...other }) => {
	const HeadingTagEl = headingTag;
	return (
		<header className="accentHeaderWrapper">
			<HeadingTagEl
				className={cx(
					`accentHeader
					accentHeader--${align}
					align--${align}
					text--normal`,
					className
				)}
			>
				{children}
			</HeadingTagEl>
		</header>
	);
};

AccentHeader.defaultProps = {
	align: 'left',
	headingTag: 'h2',
};

AccentHeader.propTypes = {
	/** Controls text alignment */
	align: PropTypes.oneOf(ALIGN_VALUES),

	/** Which level of HTML heading tag to use */
	headingTag: PropTypes.oneOf(['h1', 'h2', 'h3', 'h4', 'h5', 'h6']),
};

export default AccentHeader;
