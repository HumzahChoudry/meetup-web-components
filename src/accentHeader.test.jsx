import React from 'react';
import { shallow } from 'enzyme';

import AccentHeader, { ALIGN_VALUES } from './AccentHeader';

describe('AccentHeader', () => {
	const accentHeaderComponent = shallow(<AccentHeader>Test</AccentHeader>);

	it('exists', () => {
		expect(accentHeaderComponent).toMatchSnapshot();
	});

	it('renders classname correct alignments', () => {
		it('applies correct alignment className to dropdown content', () => {
			Object.keys(ALIGN_VALUES).forEach(alignment => {
				const accentHeaderComponent = shallow(
					<AccentHeader align={alignment}>Test</AccentHeader>
				);
				expect(accentHeaderComponent)
					.prop('className')
					.toContain(`accentHeader--${alignment}`);
			});
		});
	});

	it('renders with whatever HTML tag is passed in', () => {
		const accentHeaderComponent = shallow(
			<AccentHeader headingTag={'h1'}>Test</AccentHeader>
		);
		expect(accentHeaderComponent.find('h1').length).toBe(1);
	});
});
