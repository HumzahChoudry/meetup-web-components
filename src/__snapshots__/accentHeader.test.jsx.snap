// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccentHeader exists 1`] = `
<header
  className="accentHeaderWrapper"
>
  <h2
    className="accentHeader
					accentHeader--left
					align--left
					text--normal"
  >
    Test
  </h2>
</header>
`;
